### ISSUE TITLE
Windows Phone 7 Issue

### ISSUE THREAD

**Issue by jcwebhole** [flesler/jquery.scrollTo#56]
I am using jq1.10.2, jq-scrollto, jq-nicescroll and some custom jquery to create a single page, responsive wordpress theme.

I am almost done and ready to release the theme, here's a demo http://swgserv.com/web/j/akshaykumar/ it works on all major browsers down to ie7... it works on screens below 800px wide [yeah just 2 stops for now]..

Only problem, which a client pointed out is IE Mobile support - it displays fine but menu which is .scrollto powered isnt. I dont own a test device so I just relied on a video sent to me. Hope you guys can help me out, I tried validating the code as much as I can against HTML5, UTF-8 W3C standards still got some errors but those are from wp plugins and some inner content.

Thanks

**Comment by colir** [comment]
I've a windows phone under 8.1, and i've the same problem.
IE on windows phone need to animate the 'html'

like this $('html').animate({ scrollTop: desti }, 250);	

How to solve this with your plugin ?

actually i  use it like this
$.scrollTo( desti, scrollDuration, { easing:'easeOutSine'});

and this doesn't work under my windows phone

**Comment by flesler** [comment]
Hi @colir 

There're so many things that could happening.
- Have you tried @tmorehouse solution?
- Are you using that same page on other platforms and it works in them?
- Do you get any errors?

**Comment by colir** [comment]
Hello.
Yes i've tried @tmorehouse solution. This doesnt solve the problem
Yes, scrollto work on all device and all browser (except Windows phone) without having any error

**Comment by flesler** [comment]
I see, as I said, I don't have a WPhone myself so can't quite test solutions. If someone decyphers what is that browser expecting I'll try to handle the case specially.

**Comment by colir** [comment]
Hi. i can just tell you that to have a working scroll on my windows phone, i'm doing this 

```
if(!isIEMobile){
	$.scrollTo( desti, scrollDuration, { easing:'easeOutSine'});
}else{
	$('html').animate({scrollTop:desti},scrollDuration,'easeOutSine'})
}

```

++

**Comment by flesler** [comment]
Great, how are detecting IE mobile?

**Comment by colir** [comment]
like this
```
var isIEMobile = isIEMobile()
function isIEMobile() {
    var regExp = new RegExp("IEMobile", "i");
    return navigator.userAgent.match(regExp);
}
```
don't know his bulletproof...

**Comment by flesler** [comment]
Can you try the tests I setup at http://flesler.github.io/jquery.scrollTo/tests ?

Let me know which ones work and which don't and also include the useragent that the tests show.

**Comment by colir** [comment]
yop.
The user agent show in all your test are
Mozilla/5.0 (Mobile; Windows Phone 8.1; Android 4.0; ARM; Trident/7.0; Touch; rv:11.0; IEMobile/11.0; NOKIA; Lumia 735) like iPhone OS 7_0_3 Mac OS X AppleWebKit/537 (KHTML, like Gecko) MobileSafari/537

**Max window scrolling (quirks mode) => DONT WORKS**

**Max window scrolling (compat mode) => WORKS**

**Max element scrolling (compat mode) => WORKS**

**Max element scrolling (quirks mode) => WORKS**

**Max window scrolling, within an iframe (compat mode) => DONT WORKS**

**Max window scrolling, within an iframe (quirks mode) => WORKS**

**Max iframe scrolling from outside (compat mode) => PARTIAL (no animation)**

**Max iframe scrolling from outside (quirks mode) => PARTIAL (no animation)**

hope this help

**Comment by markvantilburg** [comment]
If you changed the 'website preference' to Desktop would that help? Or do you have the same issues then. The user agent might differ in that case.

**Comment by flesler** [comment]
I just commited a change to the https://github.com/flesler/jquery.scrollTo/tree/window-scroll branch, which contains a new beta version to fix Chrome 40, it now also (hopefully) fixes Windows Phone. 
Can you retry the tests? they contain the new beta version.

Even if this works, I'm reluctant to simply merge it to master, it'd be great to have more people test these to see there's no regression.

**Comment by colir** [comment]
hi,
i try with this minified version
https://github.com/flesler/jquery.scrollTo/blob/window-scroll/jquery.scrollTo.min.js

...but it doesn't works...

**Comment by flesler** [comment]
Hi,

Do the tests work for you now? or still the same.

**Comment by colir** [comment]
The test case suit have still the same results.

**Comment by flesler** [comment]
That is very strange, could it be that your browser is caching the JS? try clearing the cache.
It should be always using the <html> which is what you said that works.

**Comment by markvantilburg** [comment]
On my windows phone 8.1 emulator these do not work in 1.5.2beta:

Max window scrolling (compat mode)
Error message: "FAIL: scrollable must always return exactly 1 element."
Max window scrolling (quirks mode)
Error message: "FAIL: scrollable must always return exactly 1 element."
Max element scrolling (compat mode)
Error message: "FAIL: scrollable must always return exactly 1 element."
Max element scrolling (quirks mode)
Error message: "FAIL: scrollable must always return exactly 1 element."
Max iframe scrolling from outside (compat mode) => PARTIAL (no animation)
Error message: "FAIL: scrollable must always return exactly 1 element."
Max iframe scrolling from outside (quirks mode) => PARTIAL (no animation)
Error message: "FAIL: scrollable must always return exactly 1 element."


![phone_emu_agent](https://cloud.githubusercontent.com/assets/4510245/6399407/82eb38c2-bdf2-11e4-88d3-4379b9a2eac3.png)

**Comment by flesler** [comment]
Oh man, that's even worse. I don't see how the current code on _scrollable() could return more than a single elements, unless the original jQuery object had more than one element to start.

Could you get some more insights into that? I'm a bit lost.

**Comment by markvantilburg** [comment]
I'll check what happens on a real device tomorrow.

**Comment by markvantilburg** [comment]
real phone, as you can set 2 modes on the mobile phone, i tried these both:

"mobile phone mode"
All except iframe fail

Mozilla/5.0 (Mobile; Windows Phone 8.1; Android 4.0; ARM; Trident/7.0; Touch; rv:11.0; IEMobile/11.0; NOKIA; Lumia 630) like iPhone OS 7_0_3 Mac OS X AppleWebKit/537 (KHTML, like Gecko) Mobile Safari/537
document.compatMode is "CSS1Compat"
scrolling the HTML

"desktop mode"
Mozilla/5.0 (Windows NT 6.2; ARM; Trident/7.0; Touch; rv:11.0; WPDesktop; Lumia 630) like Gecko
document.compatMode is "CSS1Compat"

**Comment by flesler** [comment]
I just realized the useragent for Windows Phone has "AppleWebKit" so the first `if` was the one being used. I just reversed the order so that IEMobile is first.

Can you retest?

**Comment by markvantilburg** [comment]
On mobile with this user agent all work now

![mobile_retest_1 5 2-beta](https://cloud.githubusercontent.com/assets/4510245/6412795/ba19fad6-be88-11e4-8a8b-02ab2237cc62.png)

**Comment by flesler** [comment]
Great! still this shows the branch needs a lot of testing before releasing. A lot of browsers, other than Safar itself, include AppleWebKit in the userAgent.

**Comment by flesler** [comment]
Should I close this one?

**Comment by markvantilburg** [comment]
Well someone needs to check an actual 7.8 device, as i only checked 8.1 devices and emulators

**Comment by colir** [comment]
hi,
Sorry i was missing this day.
Where are you in the test ? how i can continue to help you ?

**Comment by markvantilburg** [comment]
If you could check if the issue is solved on WP7.8 on this url:
http://flesler.github.io/jquery.scrollTo/tests

Then the windows phone issues are solved. They are for windows phone 8 (both phone and desktop mode)

**Comment by flesler** [comment]
Hi guys,

I'd like to unify the effort of fixing Chrome 40, WP and Safari 6 into one issue so all users can test the beta version on their available browsers.
Can you please visit the issue #101, run the tests on your available browsers (desktop and mobile) and post your findings?

Thank you!

**Comment by flesler** [comment]
Hi all,
I found an interesting alternative approach mentioned on StackOverflow and decided to try it as well. It delegates much of the getting and setting scroll positions logic to jQuery's core.

It is already implemented on the [window-scroll-alt](https://github.com/flesler/jquery.scrollTo/tree/window-scroll-alt) branch. I replicated [the tests](http://flesler.github.io/jquery.scrollTo/tests2/) using this new version so people can test it too. If you comment with results, please do that on #107.

**Comment by flesler** [comment]
I merged 2.0.0 to master via #109 and this issue should be fixed.
If the problem persists please reopen the issue with details on the situation.
If you have any issue after the update, first check [this link](https://github.com/flesler/jquery.scrollTo/wiki/Migrating-to-2.0), If your problem is not solved then go ahead and [report the issue](https://github.com/flesler/jquery.scrollTo/issues/new).
